engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sPodsResourceQuotaChecker.scala

```diff
@@ -32,12 +33,13 @@ object K8sPodsResourceQuotaChecker extends LazyLogging {
     val hardAmount = podResourceQuotaOf(status.map(_.hard))
     val currentDeploymentCount = BigDecimal(oldDeploymentReplicasCount.getOrElse(0))
     val requestedReplicasCount = BigDecimal(replicasCount)
-    val quotaExceeded = (usedAmount + requestedReplicasCount) > (hardAmount + currentDeploymentCount)
+    val quotaExceeded = (usedAmount - currentDeploymentCount + requestedReplicasCount) > hardAmount
     logger.trace(s"Scenario deployment resource quota exceed: $quotaExceeded, usedPods: $usedAmount, hardPods: $hardAmount, replicasCount: $requestedReplicasCount, currentScenarioDeploymentCount: $currentDeploymentCount")
 
-    quotaExceeded match {
-      case false => Future.successful(quotaExceeded)
-      case true => throw ResourceQuotaExceededException("Quota limit exceeded")
+    if (quotaExceeded) {
+      Validated.Invalid(ResourceQuotaExceededException("Quota limit exceeded"))
```

Comment:
IMO it is more readable when Invalid  and Valid options is created in the same manner e.g:
```
if (quotaExceeded) {
  Invalid(...)
}
  Valid(Unit)
}
```

Reply:
unified